mutt_attach.c

```diff
@@ -681,15 +681,11 @@ int mutt_view_attachment(FILE *fp, struct Body *a, enum ViewAttachMode mode,
 
   if (!entry || !entry->xneomuttkeep)
   {
-    if (fp && !mutt_buffer_is_empty(tmpfile))
+    if ((fp && !mutt_buffer_is_empty(tmpfile)) || unlink_tempfile)
```

Comment:
Probably best to rename `unlink_tempfile` to something like `has_tempfile` since it no longer controls unlinking. Don't want to mislead anyone in any future debugging.

Reply:
Done